python/pyarrow/tests/test_array.py

```diff
@@ -116,9 +116,6 @@ def test_to_numpy_zero_copy():
     old_refcount = sys.getrefcount(arr)
 
     np_arr = arr.to_numpy()
-    np_arr[0] = 1
-    assert arr[0] == 1
-
     assert sys.getrefcount(arr) == old_refcount
```

Comment:
I am not fully sure what this was testing, but this line is what's failing the tests now (I suppose because `np_arr` is now referencing `arr`)

Also, is there another way to test that it was actually done zero copy? (the writing to it and see if it was updated no longer works)

Reply:
You could check the raw addresses:
```python
>>> a = pa.array([1,2,3])                                                                                                                                         
>>> buf = a.buffers()[1]                                                                                                                                          
>>> buf.address                                                                                                                                                   
140489076547648
>>> arr = a.to_numpy()                                                                                                                                            
>>> arr.ctypes.data                                                                                                                                               
140489076547648

```